crates/core/src/db/relational_db.rs

```diff
@@ -283,28 +283,50 @@ impl RelationalDB {
         };
 
         if let Some(durability) = &self.durability {
-            let inserts = tx_data
+            let inserts: Box<_> = tx_data
                 .inserts()
                 .map(|(table_id, rowdata)| Ops {
                     table_id: *table_id,
                     rowdata: rowdata.clone(),
                 })
                 .collect();
-            let deletes = tx_data
+            let deletes: Box<_> = tx_data
                 .deletes()
                 .map(|(table_id, rowdata)| Ops {
                     table_id: *table_id,
                     rowdata: rowdata.clone(),
                 })
                 .collect();
 
+            // Avoid appending transactions to the commitlog which don't modify
+            // any tables.
+            //
+            // An exception ar connect / disconnect calls, which we always want
+            // paired in the log, so as to be able to disconnect clients
+            // automatically after a server crash. See:
+            // [`crate::host::ModuleHost::call_identity_connected_disconnected`]
+            //
+            // Note that this may change in the future: some analytics and/or
+            // timetravel queries may benefit from seeing all inputs, even if
+            // the database state did not change.
+            let is_noop = || inserts.is_empty() && deletes.is_empty();
+            let is_connect_disconnect = |ctx: &ReducerContext| {
```

Comment:
This thought is sort of unrelated to this commit, but I'll just note that we should really make sure these reducers are not callable from outside of SpacetimeDB. I don't actually know that they aren't.

Reply:
Good catch ... "dunder" reducers are callable like any other reducer afaics. Because, they are just like any other reducer?